### ISSUE TITLE
Possible Glitch Where the Player Loses Y Height When Spamming Command

### ISSUE THREAD

**Issue by Toydotgame** [Toydotgame/Thisway#1]
In vanilla Minecraft 1.6, spamming `/tp @p ~100 ~ ~` (Or similar!) multiple times will cause the player to go down in Y height as they go along. This can be demonstrated using a player-size (1x2) tunnel that's multiple hundreds of blocks long and teleporting along it using `/tp @p ~100 ~ ~`. Eventually, the player's head goes into the floor; and it will continue like this as far down as it can go, presumably until the player falls into the void.

I don't know how similar the Bukkit teleportation system is to Minecraft's. If it's the same [or very similar], I'll have to add a Y height compensator which teleports up the player a very small height each time they use the command (i.e: _0.1_ or _0.2_ blocks), just enough to keep them level; and not too much to put them higher upon teleporting.

Only testing will tell!

**Comment by Toydotgame** [comment]
Research has been done, and this issue is no more!
Thankfully, it appears that Bukkit/Spigot uses a different teleportation system. When I spam the command (In this case, `/tw 100`), Y height does not sink into the ground. If `/thisway` is run for a distance outside of the loaded chunks, the player height will sink \[due to the client thinking the unloaded chunk is air, and making the player fall (In my testing, I fell from Y:50 to Y:~49.59, but the total fall height depends on how long the chunk takes to load)\]. When the chunk loads, the player is rubber-banded back to the correct coordinates and Y height.

It appears that Vanilla Minecraft's teleportation system is not too close to the server (I blame this on the cursed Release 1.3 update (Which, if you don't know; made every single player world run on an internal server - which 1. ...Made _Open to LAN_ a feature; and, 2. ruined mods, plugins, servers, clients, and the damage tilt!); whereas Bukkit's teleportation system is _part of my Spigot server_, and it works closely with it.

TL;DR: Minecraft doesn't rubber-band you when you fall into an unloaded chunk, you fall into the block when it loads; Bukkit/Spigot _does_ rubber-band you, and you don't glitch out.
